### PR TITLE
UX: Introduces a splash screen behind a hidden site setting

### PR DESCRIPTION
**Pull request by hnb-ku**
This PR introduces a new hidden site setting that allows admins to display a splash screen while site assets load.

The splash screen can be enabled via the `splash_screen` hidden site setting.

This is what the splash screen currently looks like

https://d11a6trkgmumsb.cloudfront.net/original/4X/5/c/e/5ceb72f0853a111469ca08a61328b01b39806d47.mp4

Once site assets load, the splash screen is automatically removed.

To control the `loading` text that shows in the splash screen, you can change the `preloader_text` translation string in `admin > customize > text`

### CODE REVIEW

**Comment by davidtaylorhq** [comment]
Given that we want this to show ASAP, I'm wondering if we should inline the SVG and the CSS in the page's HTML? That way, we don't need to wait for multiple network round-trips 🤔

**Comment by hnb-ku** [comment]
> Given that we want this to show ASAP, I'm wondering if we should inline the SVG and the CSS in the page's HTML? That way, we don't need to wait for multiple network round-trips thinking

Inlining the SVG won't work. It needs to either be the `src` of an `` tag or set as the `background: URL()` with CSS.

Inlining the CSS is possible. We'd lose out on the ability to immediately set the HTML `background` to a theme variable. However, I recall that @SamSaffron preferred that we have as much CSS as possible in an external stylesheet.

**Comment by SamSaffron** [comment]
Now that the other 2 are merged, can we swing this one :) 

@davidtaylorhq keep in mind that inlining here is a wash compared to the bigger 10 second delay on android, I prefer to land this first and iterate later on to see if we can show stuff even earlier. Not categorically against inlining everything, but would prefer to get something working in first and then iterate.

File: config/site_settings.yml [hunk 1]
```diff
@@ -2423,6 +2423,11 @@ uncategorized:
     default: true
     hidden: true
 
+  splash_screen:
+    default: false
+    client: true
+    hidden: true
```
Comment (davidtaylorhq) [inline config/site_settings.yml:2429]: I don't think we're actually using this on the client, so

```suggestion
  splash_screen:
    default: false
    hidden: true
```

**Comment by davidtaylorhq** [comment]
> Inlining the SVG won't work. It needs to either be the src of an  tag or set as the background: URL() with CSS.

Ahh I see. I guess we could base64-encode it, and stick it in the ` I prefer to land this first and iterate later on

Ok, sounds good - we can look at inlining later if needed 👌

**Comment by SamSaffron** [comment]
cool we have consensus then ... rebase away @hnb-ku and we can merge this in!

**Review by SamSaffron — APPROVED** [review]

**Pull request merged**